mintpy/utils/utils1.py

```diff
@@ -719,7 +719,8 @@ def run_deramp(fname, ramp_type, mask_file=None, out_file=None, datasetName=None
         print('reading data ...')
         data = readfile.read(fname)[0]
         print('estimating phase ramp ...')
-        data = deramp(data, mask, ramp_type=ramp_type, metadata=atr)[0]
+        for i in range(data.shape[0]):
+            data[i,:] = deramp(data[i,:], mask, ramp_type=ramp_type, metadata=atr)[0]
```

Comment:
The entire dataset seems to have been still loaded in to memory. It would be nice to load only what is needed to memory (which is one date at a time) do the computation, write that date and then go to the next. Otherwise this wouldn't be still memory friendly even for multi-looked data.

Reply:
I see. Thanks for the comment, and I will try to modify it.